### PR TITLE
Java Answers

### PR DESCRIPTION
**Pull request by mykeels**

### CODE REVIEW

**Comment by Thecarisma** [comment]
@mykeels  the solution is complete kindly review the PR. You can run the answers in either Netbeans or IntelliJ

**Comment by mykeels** [comment]
Thanks @Thecarisma !

I'm downloading OpenJDK right now, so I can test the solutions. Should be done tonight if NEPA is kind. 😄

File: answers/Java/@thecarisma/N02Lottery.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+/**
+ *
+ * @author Azeez Adewale <azeezadewale98@gmail.com>
+ */
+public class N02Lottery {
+    
+    public static void main(String[] args){
+        try {
```
Comment (mykeels) [inline answers/Java/@thecarisma/N02Lottery.java]: The program is supposed to repeat itself, so perhaps, have a while loop around the try {...} ?

File: answers/Java/@thecarisma/N04TypeInReverse.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    public static void main(String[] args){
+        Scanner scanner = new Scanner(System.in);
+        String word, userInput ;
+        
+        while (true) {
+            word = getRandomWord();
+            System.out.printf("Type the word '%s' in reverse: ", word);
+            userInput = scanner.nextLine();
+            if (userInput.equals(reverseWord(word))) 
+                System.out.println("✅");
```
Comment (mykeels) [inline answers/Java/@thecarisma/N04TypeInReverse.java]: I get an error saying `"error: unmappable character (0x9D) for encoding windows-1252"`

See https://stackoverflow.com/a/464886/7396801

**Comment by Thecarisma** [comment]
I have resolve the issue with the Unicode character in question 4 and the logic in question 2 has been wrapped in a while loop as suggested

**Comment by mykeels** [comment]
Now, for the N02Lottery problem, I get two responses when I press enter.

image

**Comment by mykeels** [comment]
For N04TypeInReverse, I get a question mark, in place of the emojis when I enter a response.

image

**Comment by mykeels** [comment]
For N14NumberInWords, there's an `and` missing from the `hundreds` words. 

E.g. `Three hundred "and" forty five`

image

**Comment by Thecarisma** [comment]
For the N04TypeInReverse the issue is the JDK you have does not support Unicode encoding which means the first error should not occur in the first place, I replaced the emojis with their Unicode value so instead of the error it shows the question mark. You can try to run the solution in an Online Java Compiler. 

For N02Lottery 🙈, am using System.in directly instead of Scanner. Try another terminal for N02Lottery  if the error persist

I committed changes for N02Lottery

**Comment by Thecarisma** [comment]
> For N14NumberInWords, there's an `and` missing from the `hundreds` words.
> 
> E.g. `Three hundred "and" forty five`
> 

Changes committed https://github.com/mykeels/20-questions/pull/8/commits/164bbfa55be0625e6c36a3dbd5a2c32c060c174f

**Comment by mykeels** [comment]
For `N16ConvetToDecimal`, the file name has a misspelling, and should be `N16ConvertToDecimal` instead.

Also, the output phrase `7 in base 2 is 111 in base 10` should be `7 base 10 is 111 base 2`. 

> The previous is incorrect because it suggests that the number 7 is in base 2.

**Comment by mykeels** [comment]
For `N17ConvertDecimalToBaseN`, the program should request a decimal value, then convert to base `n`, not the other way around.

Also, the output phrase, `1000101 in base 10 is 69 in base 2` should be `1000101 base 2 is 69 base 10.

---

For `N18SimpleCGPACalculator`, the average and cgpa output should be floating point values.

---

For `N20LoveCalculator`, I get `error: package com.sun.xml.internal.ws.util does not exist`.

**Comment by Thecarisma** [comment]
> For `N17ConvertDecimalToBaseN`, the program should request a decimal value, then convert to base `n`, not the other way around.
> 
> Also, the output phrase, `1000101 in base 10 is 69 in base 2` should be `1000101 base 2 is 69 base 10.
> 
> For `N18SimpleCGPACalculator`, the average and cgpa output should be floating point values.
> 
> For `N20LoveCalculator`, I get `error: package com.sun.xml.internal.ws.util does not exist`.

Committed changes https://github.com/mykeels/20-questions/pull/8/commits/64f35dd7168772561cc659518b73268473df2d05

**Comment by mykeels** [comment]
For N16ConvertToDecimal, here's the output:

image

Notice that both base 2 and 10 values are in base 2.

I believe this is the last one.

**Pull request merged**